stable/democratic-csi/templates/node.yaml

```diff
@@ -266,8 +267,10 @@ spec:
       {{- if $mountIscsi }}
       - name: iscsi-dir
         hostPath:
-          path: /etc/iscsi
+          path: {{ .Values.node.driver.iscsiDirHostPath }}
+          {{- if .Values.node.driver.iscsiDirHostPathCheckDirectory }}
```

Comment:
What is the behavior of talos that deems this necessary?

Reply:
talos uses `/usr/local/etc/iscsi` and not `/etc/iscsi` since everything in `/etc` is read-only.

As for the `iscsiDirHostPathCheckDirectory` I tried every possible value in here https://kubernetes.io/docs/concepts/storage/volumes/#hostpath except the ones with `orCreate` and I couldint get it to pass the check. Leaving it empty made it work. I'm not proud of this. Maybe because the `/usr/local/etc/iscsi` is a mount from a docker container (the extension) inside the talos node the checks fails, I verified that it is a dictionary and accessable by bash-ing into the container with the `type` check empty and it exist.

if you have a guess of why it is like this let me know and I'm happy to test it.

Reply:
Fair enough, it's likely related to mount propagation but that's just a guess.

Reply:
I see, thank you for the hint! If you can wait for me until the next weekend I might try edit the iscsi-tools talos extension and add [rshared](https://docs.docker.com/storage/bind-mounts/#configure-bind-propagation) in [here](https://github.com/siderolabs/extensions/blob/e0b1465c5ce60aa6c761b2a75ea54a8b69f130f6/storage/iscsi-tools/iscsid.yaml#L52) in a new demo cluster since I dont feel comfortable experimenting with patches on my current cluster.
Assuming it worked (Which I hope) I would need to get my pull request there merged and versioned in a release so it can be used.

This will take time. We have two options:
* keeping `iscsiDirHostPathCheckDirectory` for now and deprecating it later once I find a solution if there is any.
* waiting until we confirm whether `rshared` solution works and decide from there.

What are your thoughts?

Reply:
Yeah that’s fine. Although it’s probably better to just make the value configurable vs conditional (and then you can just change the value to the k8s default instead of chart default).

Reply:
Good idea, I did it as a `iscsiDirHostType` and used `Directory` as the default. Will consider going after talos rshared thing as future work.